Software/bsw/static/Mcal/CAN/src/Can.c

```diff
@@ -830,63 +905,54 @@ void Can_DeInit(void)
 void Can_MainFunction_Read(void) 
 {
 
+/*variable to get HRH index in the HWObject array */
+ uint8 index = 0 ;
+/*array to save received data*/
+ uint8 data[MAX_DATA_LENGTH] ;
+
     /*
-     *   Loop all controllers to get the new data
+     * loop for all the hardware object to get the only new avialables data in this object
      */
-    // TODO controlledID  base address not implemented here ?
-    uint8 controllerId = 0 ; /*variable to count controllers number*/
-    for (controllerId = 0; controllerId < USED_CONTROLLERS_NUMBER; controllerId++)
+
+    for(uint8 obj_index = 0; obj_index < CAN_HRH_NUMBER; obj_index++)
     {
-        /*SWS_Can_00108*/
-        if ((POLLING_PROCESSING == Global_Config->CanHardwareObjectRef[controllerId].CanControllerRef->CanRxProcessing) ||
-            (
-                    (MIXED_PROCESSING == Global_Config->CanHardwareObjectRef[controllerId].CanControllerRef->CanRxProcessing) &&
-                    (TRUE == Global_Config->CanHardwareObjectRef[controllerId].CanHardwareObjectUsesPolling)
-            ))
-        {
-            uint8 obj_index;  /* variable to count object number */
-#if (CAN_DEV_ERROR_DETECT == STD_ON)
-        if (ControllerState[controllerId] == CAN_CS_UNINIT)
-        {
-                Det_ReportError( CAN_MODULE_ID, CAN_INSTANCE_ID, CAN_MAIN_FUNCTION_READ_ID, CAN_E_UNINIT);
-        }//End if
-#endif //CanDevErrorDetect
-        /*
-         * loop for all the hardware object to get the only new avialables data in this object
-         */
-        // TODO psMsgObject shold be config inside init API
-        for(obj_index = 0; obj_index < NUM_OF_HOH; obj_index++)
-        {
-            if(Global_Config->CanHardwareObjectRef[obj_index].CanObjectType == RECEIVE)
+        index = MessageObjAssignedToHRH[obj_index].HRHIndex ;
+       if(TRUE == Global_Config->CanHardwareObjectRef[index].CanHardwareObjectUsesPolling)
```

Comment:
It should check if CanRxProcessing is set to POLLING or MIXED .
[SWS_Can_00108] ⌈ The function Can_MainFunction_Read shall perform the polling of RX indications when CanRxProcessing is set to POLLING or MIXED.
In case of MIXED , then check for CanHardwareObjectUsesPolling .

Reply:
For me I can see this as:
- Interrupt -> No main function
- Polling or mixed : 
    -> if mixed: check if the msg configured for polling.
        -> if polling: check
        -> else -> continue

Reply:
We agree on the current implementation of  looping over all the msgs, we check if it's polling or not, and check the msg based on that.